logger.go

```diff
@@ -21,5 +22,22 @@ func NewLogger(w io.Writer) Logger {
 		w = os.Stderr
 	}
 	enc := json.NewEncoder(w)
-	return LoggerFunc(func(event Event) error { return enc.Encode(event) })
+	return LoggerFunc(func(event Event) error { return encode(enc, event) })
+}
+
+func encode(enc *json.Encoder, event Event) (err error) {
+	if err = enc.Encode(event); err == nil {
+		return
+	}
+
+	// Attempts to recover from invalid data put in the free form Event.Data field.
+	switch err.(type) {
+	case *json.UnsupportedTypeError, *json.UnsupportedValueError, *json.MarshalerError:
+		event.Level = ALERT
+		event.Info.Errors = append(event.Info.Errors, MakeEventError(err))
+		event.Data = EventData{"unserializable": fmt.Sprintf("%#v", event.Data)}
```

Comment:
The only limit I'm aware of is 1 MB per event batch when sending to CloudWatchLogs... I don't think we'll ever reach this with a single event tho.